FWIO/RNTuple/plugins/RNTupleOutputFile.cc

```diff
@@ -0,0 +1,583 @@
+                                       Config const& iConfig,
+                                       bool anyProductProduced)
+      : file_(iFileName.c_str(), "recreate", ""),
+        wrapperBaseTClass_(TClass::GetClass("edm::WrapperBase")),
+        selectorConfig_(iConfig.selectorConfig),
+        dropMetaData_(iConfig.dropMetaData) {
+    setupRuns(iSelected[InRun], iConfig);
+    setupLumis(iSelected[InLumi], iConfig);
+    setupEvents(iSelected[InEvent], iConfig, anyProductProduced);
+    setupPSets(iConfig);
+    setupParentage(iConfig);
+    setupMetaData(iConfig);
+
+    auto const& branchToChildMap = iFileBlock.productDependencies().childLookup();
+    for (auto const& parentToChildren : branchToChildMap) {
+      for (auto const& child : parentToChildren.second) {
+        productDependencies_.insertChild(parentToChildren.first, child);
+      }
+    }
+  }
+
+  namespace {
+    std::string fixBranchName(std::string const& iName) {
+      //need to remove the '.' at the end of the branch name
+      return iName.substr(0, iName.size() - 1);
+    }
+
+    void noSplitField(ROOT::Experimental::RFieldBase& iField) {
+      auto const& typeName = iField.GetTypeName();
+      if (typeName == "std::uint16_t") {
+        iField.SetColumnRepresentatives({{ROOT::Experimental::EColumnType::kUInt16}});
+      } else if (typeName == "std::uint32_t") {
+        iField.SetColumnRepresentatives({{ROOT::Experimental::EColumnType::kUInt32}});
+      } else if (typeName == "std::uint64_t") {
+        iField.SetColumnRepresentatives({{ROOT::Experimental::EColumnType::kUInt64}});
+      } else if (typeName == "std::int16_t") {
+        iField.SetColumnRepresentatives({{ROOT::Experimental::EColumnType::kInt16}});
+      } else if (typeName == "std::int32_t") {
+        iField.SetColumnRepresentatives({{ROOT::Experimental::EColumnType::kInt32}});
+      } else if (typeName == "std::int64_t") {
+        iField.SetColumnRepresentatives({{ROOT::Experimental::EColumnType::kInt64}});
+      } else if (typeName == "float") {
+        iField.SetColumnRepresentatives({{ROOT::Experimental::EColumnType::kReal32}});
+      } else if (typeName == "double") {
+        iField.SetColumnRepresentatives({{ROOT::Experimental::EColumnType::kReal64}});
+      }
+    }
+
+    void checkSubFieldsForNoSplit(ROOT::Experimental::RFieldBase& iField,
+                                  std::vector<std::string> const& iNoSplitFields) {
+      for (auto const& name : iNoSplitFields) {
+        if (name.starts_with(iField.GetFieldName())) {
+          bool found = false;
+          for (auto& subfield : iField) {
+            if (subfield.GetQualifiedFieldName() == name) {
+              found = true;
+              noSplitField(subfield);
+              break;
+            }
+          }
+          if (not found) {
+            edm::LogError("MissingSubField")
+                << "The data product was found but the requested subfield '" << name << "' is not part of the class";
```

Comment:
Should we throw an exception instead? If not, some explanation of why not would be nice (in comments).

Reply:
Throwing probably makes the most sense.